protocol/sync_info.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  IDENTIFIER_LENGTH,
+  identifierFromBytes,
+  identifierToBytes,
+} from "../chain/identifier.ts";
```

Comment:
move the branded uint8array factory to a shared module and make `Identifier` use that as well